dietpi-install.sh

```diff
@@ -1,38 +1,40 @@
 #!/bin/bash
 
 # Variables
-IMAGE_URL=$(whiptail --inputbox "Enter the URL for the DietPi image (default: https://dietpi.com/downloads/images/DietPi_Proxmox-x86_64-Bullseye.7z):" 8 78 https://dietpi.com/downloads/images/DietPi_Proxmox-x86_64-Bullseye.7z --title "DietPi Installation" 3>&1 1>&2 2>&3)
-RAM=$(whiptail --inputbox "Enter the amount of RAM (in MB) for the new virtual machine (default: 2048):" 8 78 2048 --title "DietPi Installation" 3>&1 1>&2 2>&3)
-CORES=$(whiptail --inputbox "Enter the number of cores for the new virtual machine (default: 2):" 8 78 2 --title "DietPi Installation" 3>&1 1>&2 2>&3)
+IMAGE_URL=$(whiptail --inputbox 'Enter the URL for the DietPi image (default: https://dietpi.com/downloads/images/DietPi_Proxmox-x86_64-Bullseye.7z):' 8 78 'https://dietpi.com/downloads/images/DietPi_Proxmox-x86_64-Bullseye.7z' --title 'DietPi Installation' 3>&1 1>&2 2>&3)
+RAM=$(whiptail --inputbox 'Enter the amount of RAM (in MB) for the new virtual machine (default: 2048):' 8 78 2048 --title 'DietPi Installation' 3>&1 1>&2 2>&3)
+CORES=$(whiptail --inputbox 'Enter the number of cores for the new virtual machine (default: 2):' 8 78 2 --title 'DietPi Installation' 3>&1 1>&2 2>&3)
 
 # Get the next available VMID
 ID=$(pvesh get /cluster/nextid)
 
-touch /etc/pve/qemu-server/$ID.conf
+touch "/etc/pve/qemu-server/$ID.conf"
 
 # Get the storage name from the user
-STORAGE=$(whiptail --inputbox "Enter the storage name where the image should be imported:" 8 78 --title "DietPi Installation" 3>&1 1>&2 2>&3)
+STORAGE=$(whiptail --inputbox 'Enter the storage name where the image should be imported:' 8 78 --title 'DietPi Installation' 3>&1 1>&2 2>&3)
 
 # Download DietPi image
-wget $IMAGE_URL
+wget "$IMAGE_URL"
 
 # Extract the image
-7zr x DietPi_Proxmox-x86_64-Bullseye.7z
+IMAGE_NAME=${IMAGE_URL##*/}
+IMAGE_NAME=${IMAGE_NAME%.7z}
+7zr e "$FILE_NAME.7z" "$FILE_NAME.qcow2"
```

Comment:
An error is thrown on this line when running tests on my system. Unfortunately this is beyond my knowledge. Have rolled back to standard extract until i can understand what is going on. Are you able to help, @MichaIng

Reply:
Ah my fault, it should be `IMAGE_NAME` here as well, of course.

Reply:
Fixed with: #4